### PR TITLE
[tx] Optimize accumulate_grads with global JIT-compiled accumulator

### LINKED ISSUE #647: [tx] Improve performance of accumulate_grads
Currently (after https://github.com/NovaSky-AI/SkyRL/pull/642) if you look at the flame graph

Image

a decent amount of time is spent in accumulate_grads. Ideally we can make this part of the forward_backward pass jit, so the overhead can be reduced (in our past experience this is very effective). It would be best if we can retain most of the structure of the current code, maybe that could be done by annotating `AccumulatedGradients` with the `@jax.tree_util.register_dataclass` decorator.

### PR DESCRIPTION
**Pull request by hezyin**
# [tx] Optimize accumulate_grads with global JIT-compiled accumulator

## Summary
This PR optimizes the `accumulate_grads` function in `TinkerEngine` by refactoring `AccumulatedGradients` into a global, JIT-compiled JAX dataclass. This change moves the gradient accumulation logic from Python loops into the compiled XLA kernel, significantly reducing dispatch overhead and improving training performance.

## Changes
- **Refactored `AccumulatedGradients`**:
    - Converted to a JAX dataclass using `@jax.tree_util.register_dataclass`.
    - Now stores gradients for **all** adapters in a single structure (`grad_sum` matching `lora_params` shape) and a global `counts` array.
    - Implemented a JIT-compiled `add` method that uses `jnp.bincount` and tree mapping for efficient batch accumulation.
- **Updated `TinkerEngine`**:
    - Replaced the dictionary of per-model accumulators with a single global `AccumulatedGradients` instance.
    - Updated `_accumulate_grads` to use the new JIT-compiled `add` method.
    - Updated `process_optim_step` to extract gradients for specific adapters, apply updates, and reset the global accumulator for that adapter.
    - Updated `process_create_model` and `process_forward_backward_batch` to align with the new global accumulator strategy.

## Verification
- **Automated Tests**: Ran existing tests in `skyrl-tx/tests/tinker/test_engine.py`.
    - `test_adapter_gradient_calculation`: Passed. Confirmed that the new JIT-ed accumulation produces identical gradients to the previous implementation.
    - `test_micro_batch_grad_accumulation`: Passed. Verified that micro-batch accumulation works correctly with the new global structure.

## Related Issue
Fixes #647

### CODE REVIEW

File: skyrl-tx/tx/tinker/engine.py [hunk 1]
```diff
@@ -443,7 +459,6 @@ def process_create_model(self, model_id: str, request_data: types.CreateModelInp
             adapter_index=adapter_index,
             lora_config=lora_config,
         )
-        self.accumulated_grads[model_id] = AccumulatedGradients()
```
Comment (erkamkavak) [inline skyrl-tx/tx/tinker/engine.py:446]: Hi, just an observation: shouldnt it call reset adapter for corresponding model?
```
        self.accumulated_grads = self.accumulated_grads.reset_adapter(adapter_index)
```

**Comment by pcmoritz** [comment]
This PR is now merged as part of https://github.com/NovaSky-AI/SkyRL/pull/691, I made you a co-author @hezyin ! Thanks a lot for working on this ❤️
